File: backend/models/ollama_response.py
from pydantic import BaseModel
from typing import Optional, Dict, Any

class OllamaMessage(BaseModel):
    role: str
    content: str

class OllamaResponse(BaseModel):
    model: str
    created_at: str
    message: OllamaMessage
    done_reason: Optional[str] = None
    done: bool
    total_duration: Optional[int] = None
    load_duration: Optional[int] = None
    prompt_eval_count: Optional[int] = None
    prompt_eval_duration: Optional[int] = None
    eval_count: Optional[int] = None
    eval_duration: Optional[int] = None

File: frontend/src/types/ollamaResponse.ts
import { z } from "zod";

export const OllamaMessageSchema = z.object({
  role: z.string(),
  content: z.string(),
});

export const OllamaResponseSchema = z.object({
  model: z.string(),
  created_at: z.string(),
  message: OllamaMessageSchema,
  done_reason: z.string(),
  done: z.boolean(),
  total_duration: z.number(),
  load_duration: z.number(),
  prompt_eval_count: z.number(),
  prompt_eval_duration: z.number(),
  eval_count: z.number(),
  eval_duration: z.number(),
});

export type OllamaMessage = z.infer<typeof OllamaMessageSchema>;

export type OllamaResponse = z.infer<typeof OllamaResponseSchema>;
